free-2.md

```diff
@@ -11,41 +11,135 @@ _What need is met by your idea? Who are you helping? What is that person's
+ordinary traders. According to Prof. Evans, most commercial trading software is
+written in C, which obviously requires a technical knowledge lacked by most
+traders. Why not have a DSL specifically designed to make trading operations
+easy, and open up this technique to more than just large corporations?
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+There are so many different types of investment strategies that a user might
+want to try that a non-language application would be insufficient. Instead, we
+should provide the user with the basic functions of trading and some means for
+implementing financial logic, and let the user construct their strategy by
+combining these basic functions.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I'm currently taking Financial Econ with Prof. Evans, and in a recent lecture
+we learned about spread arbitrage and algo-trading, which I found very
+interesting. Prof. Evans was careful to point out that, by carefully selecting
+stocks which are not heavily computer-traded, a small trader could make a
+reasonable profit by implementing a trading algorithm. He even suggested that
+we Mudders should try it out.
+
+I believe that this opportunity should be open not just to Mudders and other
+technically savvy people, but to anyone with an interest in taking a more
+active role in managing their finances.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Intuitive automation for trading stocks.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I imagine the language as a set of commands that can be used to describe a portfolio and a strategy. The most basic commands might look like:
+```
+BUY <x> SHARES OF <company a>
+SELL <y> SHARES OF <company b>
+```
+
+There should also be some logic associated with these simply buy/sell orders:
+```
+BUY <x> SHARES OF <company c> IF THE PRICE FALLS BELOW <dollar amount>
+SELL <y> SHARES OF <company d> FOR <dollar amount>
+```
+
+There should also be a way to interact with the state of the market:
+```
+IF SPREAD(<company e>) IS MORE THAN <dollar amount> THEN <do something>
+```
+
+Finally, the user may not want to specify companies explicitly, but rather execute some strategy for any company that meets a criterion, like:
+```
+FOR COMPANIES WITH SPREAD <dollar amount> AND VOLUME <shares/day> <do something>
+```
+
+I've written these example commands in a natural language-like pseudo-code to
+reflect the intention to make the language accessible to nontechnical people.
+However, I do not think that natural language is the best way to go (see
+AppleScript). One of the challenges in designing the syntax for this language
+would be in striking the best balance between preciseness and readability.
```

Comment:
I agree that we should be careful in how much natural language we put into our syntax. I think this problem would benefit from at least some use of natural language.